internal/wallet/address.go

```diff
@@ -1,20 +1,20 @@
 package wallet
 
 import (
-	"github.com/adiabat/btcd/btcec"
-	"github.com/adiabat/btcd/txscript"
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcd/txscript"
 	"github.com/btcsuite/btcutil"
 	"github.com/btcsuite/btcwallet/waddrmgr"
 	"github.com/btcsuite/btcwallet/walletdb"
 )
 
 func (w *wallet) NewPubkey() (pub *btcec.PublicKey, err error) {
-	// TODO: generate new pubkey and address using newAddress
-	_, err = w.newAddress(waddrmgr.KeyScopeBIP0084, []byte{}, uint32(1), uint32(1))
+	mAddrs, err := w.newAddress(waddrmgr.KeyScopeBIP0084, w.privatePassphrase, uint32(1), uint32(1))
```

Comment:
I still wonder why you need private key, though.
I guess the manager already is unlocked? so you don't need to unlock it everytime you write.
https://github.com/btcsuite/btcwallet/blob/c4dd27e481f9801866cf5226bfe532084772ec2a/wallet/wallet.go#L2830

Reply:
I guess that a private kay was necessary for creating managers, but not for modifying records.